dashboards-reports/public/components/report_definitions/delivery/delivery.tsx

```diff
@@ -84,24 +117,107 @@ export function ReportDelivery(props: ReportDeliveryProps) {
     };
   };
 
+  const sendTestNotificationsMessage = () => {
+    // implement send message
+    // on success, set test message confirmation message
+    handleTestMessageConfirmation(testMessageConfirmationMessage)
+  }
+
   useEffect(() => {
+    // to replace with actual channels from notifications plugin
+    setChannels(placeholderChannels);
     if (edit) {
       httpClientProps
         .get(`../api/reporting/reportDefinitions/${editDefinitionId}`)
-        .then(async (response) => {
-          const isEmailSelected =
-            response.report_definition.delivery.delivery_type ===
-            DELIVERY_TYPE_OPTIONS[1].id;
-          handleEmailCheckbox({ target: { checked: isEmailSelected } });
+        .then(async (response: any) => {
+          if (response.report_definition.delivery.configIds.length > 0) {
+            // add config IDs
+            setSendNotification(true)
+            let delivery = response.report_definition.delivery;
+            let editChannelOptions = [];
+            for (let i = 0; i < delivery.configIds.length; ++i) {
+              let editChannelOption = {
+                label: delivery.configIds[i]
```

Comment:
i think you need to show the channel names instead of configIds in UI?

Reply:
The doc for API integration shows `configIds` is array of strings. If the channel names are not the configIds I'll update but will have to change the schema so that configIds is an array of objects

Reply:
the configIds should be an array of strings in the schema, but the strings will look like `07wErHoBJ4_3xKCiCrRg`. I think on UI side you need to display by channel names, and send the corresponding ids to reports scheduler

Reply:
If the front-end is receiving an array of schemas from the backend, how will I be able to correctly map the channel name to the config Id? Does the notifications API call return an array of objects with something like `{ label: string, configId: string}`?

Reply:
The backend returns both id and name for each channel, id is `channel.config_id`, name is `channel.config.name`. 

Is reporting calling notifications OpenSearch plugin or the notifications dashboards server?

Reply:
Got you, if it receives both from Notifications then we are good. I'll update the placeholder array to reflect that. For this call if it is already defined in the Notifications dashboards server I don't see why we can't just call that same API from reporting.

Reply:
actually no, in the reporting schema you only have configIds field. you'll need to do it in two calls then, one for report definition, one for channel names

Reply:
If the Notifications API returns both config ID and channel name, then that can be parsed on the Reporting dashboards side. I'm not sure I see why we would need an extra API call

Reply:
sorry you can ignore that, i was thinking for edit you need one call to get the reporting schema, one call to get channels, but that's true regardless of the name issue

Reply:
👍